metagraph/plugins/pandas/types.py

```diff
@@ -104,9 +114,22 @@ def assert_equal(
                 g1 = obj1.value
                 g2 = obj2.value
                 assert len(g1) == len(g2), f"{len(g1)} != {len(g2)}"
-                assert len(obj1.index & obj2.index) == len(
-                    obj1.index
-                ), f"{len(obj1.index & obj2.index)} != {len(obj1.index)}"
+                if aprops1["is_directed"]:
+                    assert len(obj1.index & obj2.index) == len(
+                        obj1.index
+                    ), f"edge mismatch {obj1.index ^ obj2.index}"
+                else:  # undirected
```

Comment:
It might be nice to move `g1.set_index([obj1.dst_label, obj1.src_label]).index` out to a variable so that back doesn't split this up into two lines. The same might go for g2 as well.